tasks/main.yml

```diff
@@ -116,3 +116,39 @@
   import_tasks: ./tls.yml
   tags:
     - tls
+
+- name: Upload the watcher path systemd template
+  template:
+    src : "../templates/etc/systemd/system/etc-resolv-conf-watcher.path.j2"
+    dest: "/etc/systemd/system/etc-resolv-conf-watcher.path"
+    mode: '0644'
+  when: enable_resolv_conf_watcher
+  register: path_template_result
+
+- name: Upload the watcher service systemd template
+  template:
+    src : "../templates/etc/systemd/system/etc-resolv-conf-watcher.service.j2"
+    dest: "/etc/systemd/system/etc-resolv-conf-watcher.service"
+    mode: '0644'
+  when: enable_resolv_conf_watcher
+  register: service_template_result
+
+- name: Enable the watcher path
+  systemd:
+    name: etc-resolv-conf-watcher.path
+    state: started
+    enabled: '{{ enable_resolv_conf_watcher | ternary("true", "false") }}'
+    daemon_reload: yes
+  when: enable_resolv_conf_watcher
+    and path_template_result is defined
+    and path_template_result.changed
+
+- name: Enable the watcher service
+  systemd:
+    name: etc-resolv-conf-watcher.service
+    state: started
+    enabled: '{{ enable_resolv_conf_watcher | ternary("true", "false") }}'
+    daemon_reload: yes
+  when: enable_resolv_conf_watcher
+    and service_template_result is defined
+    and service_template_result.changed
```

Comment:
@Chithrak07 we're almost there, but stepping back and having a look at this, I believe a couple more final things will need to be added:
 
* `state` also needs a ternary to toggle started/stopped
* Now that we've added the all the `when` conditions to the systemd tasks, they'll never actually fire to stop and disable the service/watcher if `enable_resolv_conf_watcher` is ever false. But obviously we need it to. There's also the added complication of passing idempotence, which makes it quite the tangled logical knot. Let me know if you have any ideas here.

Reply:
@dale-c-anderson : 
What if we check if the service is enabled and also check if the var is set to false and then we can try disabling the service?

But, Why does the state always shows 'stopped'?

Reply:
@Chithrak07 what about offloading the tasks to two separate versions of include files?

Do everything relevant to enabling the service in one version, and then doing everything relevant to disabling it in another?  Then just include one or the other based on the value of `enable_resolv_conf_watcher`? The tasks we've got already could be placed in the 'enabled' version, and then we could just write opposite operations in the 'disabled' include?

Reply:
@dale-c-anderson : Can you please review the latest changes?

Reply:
This is looking good so far, but the centos 8 test fails - can you check and see what's going on there?

Reply:
Fixed.